src/it/scala/djinni/IntegrationTest.scala

```diff
@@ -67,34 +70,36 @@ class IntegrationTest extends FunSpec {
     * @return command-line output of the executed djinni-cli
     */
   def djinniGenerate(idl: String): String = {
+    val baseOutputPath = "src/it/resources/result"
     djinni("--java-package djinni.it " +
-      s"--java-out src/it/resources/result/$idl/java " +
-      s"--cpp-out src/it/resources/result/$idl/cpp " +
-      s"--cpp-header-out src/it/resources/result/$idl/cpp-headers " +
-      s"--jni-out src/it/resources/result/$idl/jni " +
-      s"--jni-header-out src/it/resources/result/$idl/jni-headers " +
-      s"--objc-out src/it/resources/result/$idl/objc " +
-      s"--objc-header-out src/it/resources/result/$idl/objc-headers " +
+      s"--java-out $baseOutputPath/$idl/$JAVA " +
+      s"--cpp-out $baseOutputPath/$idl/$CPP " +
+      s"--cpp-header-out $baseOutputPath/$idl/$CPP_HEADERS " +
+      s"--jni-out $baseOutputPath/$idl/$JNI " +
+      s"--jni-header-out $baseOutputPath/$idl/$JNI_HEADERS " +
+      s"--objc-out $baseOutputPath/$idl/$OBJC " +
+      s"--objc-header-out $baseOutputPath/$idl/$OBJC_HEADERS " +
       "--objc-swift-bridging-header bridging-header " +
       "--objc-type-prefix IT " +
-      s"--objcpp-out src/it/resources/result/$idl/objcpp " +
-      s"--py-out src/it/resources/result/$idl/python " +
-      s"--pycffi-out src/it/resources/result/$idl/cffi " +
-      s"--c-wrapper-out src/it/resources/result/$idl/cwrapper " +
-      s"--c-wrapper-header-out src/it/resources/result/$idl/cwrapper-headers " +
-      s"--c-wrapper-include-prefix ../cwrapper-headers/ " +
-      s"--c-wrapper-include-cpp-prefix ../cpp-headers/ " +
-      s"--cppcli-out src/it/resources/result/$idl/cppcli " +
-      s"--cppcli-include-cpp-prefix ../cpp-headers/ " +
-      s"--list-out-files src/it/resources/result/$idl/generated-files.txt " +
+      s"--objcpp-out $baseOutputPath/$idl/$OBJCPP " +
+      s"--objcpp-header-out $baseOutputPath/$idl/$OBJCPP_HEADERS " +
+      s"--py-out $baseOutputPath/$idl/python " +
+      s"--pycffi-out $baseOutputPath/$idl/cffi " +
+      s"--c-wrapper-out $baseOutputPath/$idl/$CWRAPPER " +
+      s"--c-wrapper-header-out $baseOutputPath/$idl/$CWRAPPER_HEADERS " +
+      s"--c-wrapper-include-prefix ../$CWRAPPER_HEADERS/ " +
+      s"--c-wrapper-include-cpp-prefix ../$CPP_HEADERS/ " +
+      s"--cppcli-out $baseOutputPath/$idl/$CPPCLI " +
+      s"--cppcli-include-cpp-prefix ../$CPP_HEADERS/ " +
+      s"--list-out-files $baseOutputPath/$idl/generated-files.txt " +
```

Comment:
Don't be overwhelmed by these changes. I realized that I could/should make use of the existing Language variables (`CPP`, `CPP_HEADERS`, ...), which specify where to find the generated output. 
The only real change in here is the addition of
```
s"--objcpp-header-out $baseOutputPath/$idl/$OBJCPP_HEADERS " +
```

Reply:
yes, it's just we do not have any tests for the default parameter, right? 
not that this is important, I can see in the code that that should work (as long as no one changes the order of things)
but the list of command line option cases we do not test also grew :-)

Reply:
You are right about that. I added a special test case that is making sure the `--objcpp-out` parameter is working standalone. While I am aware that this is only implicitly tests the fallback, I think it will do the job. I tried to make it clear in the test description that the fallback is also something that we want to make sure...
https://github.com/cross-language-cpp/djinni-generator/blob/c4768384cf29aad06109f96fae502141b5600d90/src/it/scala/djinni/GeneratorIntegrationTest.scala#L224